src/main/java/com/google/api/codegen/transformer/ParamWithSimpleDoc.java

```diff
@@ -0,0 +1,123 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer;
+
+import com.google.api.codegen.viewmodel.ParamDocView;
+import com.google.api.codegen.viewmodel.RequestObjectParamView;
+import com.google.api.codegen.viewmodel.SimpleParamDocView;
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Function;
+import com.google.common.collect.FluentIterable;
+import java.util.List;
+import javax.annotation.Nullable;
+
+@AutoValue
+public abstract class ParamWithSimpleDoc {
+
+  public static List<RequestObjectParamView> asRequestObjectParamViews(
+      List<ParamWithSimpleDoc> ps) {
+    return FluentIterable.from(ps)
+        .transform(
+            new Function<ParamWithSimpleDoc, RequestObjectParamView>() {
+              @Override
+              public RequestObjectParamView apply(ParamWithSimpleDoc p) {
+                return p.asRequestObjectParamView();
+              }
+            })
+        .toList();
+  }
+
+  public static List<ParamDocView> asParamDocViews(List<ParamWithSimpleDoc> ps) {
+    return FluentIterable.from(ps)
+        .transform(
+            new Function<ParamWithSimpleDoc, ParamDocView>() {
+              @Override
+              public ParamDocView apply(ParamWithSimpleDoc p) {
+                return p.asParamDocView();
+              }
+            })
+        .toList();
```

Comment:
I know this is a pain, but could you convert your remaining FluentIterable usages to plain loops? Particularly in simple cases like this, the FluentIterable approach is considerably more verbose (twice the lines) as compared to plain loops:

```
List<ParamDocView> views = new ArrayList<>();
for (ParamWithSimpleDoc param : ps) {
  views.add(param.asParamDocView());
}
return views;
```

I will be quite happy when we can use the much more concise Java 8 style.